src/core/mapping/instance_manager.h

```diff
@@ -162,7 +207,31 @@ class InstanceManager {
 
  private:
   std::map<FieldMemInfo, InstanceSet> instance_sets_{};
-  Legion::Mapping::LocalLock manager_lock_{};
+};
+
+class ReductionInstanceManager : public BaseInstanceManager {
```

Comment:
Don't inherit `BaseInstanceManager`, as `find`/`record_instance` in that class don't make any sense to reduction instances.

Reply:
`BaseInstanceManager` doesn't have these methods, it is just a placeholder for the common data and functions between InstanceManager and ReductionInstanceManager